plugins/logs/mask_test.go

```diff
@@ -0,0 +1,489 @@
+		exp   string
+	}{
+		{
+			note: "erase input",
+			ptr: maskRule{
+				OP:   maskOPRemove,
+				Path: "/input",
+			},
+			event: `{"input": {"a": 1}}`,
+			exp:   `{"erased": ["/input"]}`,
+		},
+		{
+			note: "upsert input",
+			ptr: maskRule{
+				OP:   maskOPUpsert,
+				Path: "/input",
+				Value: struct {
+					RandoString string
+				}{RandoString: "foo"},
+			},
+			event: `{"input": {"a": 1}}`,
+			exp:   `{"masked": ["/input"], "input": {"RandoString": "foo"}}`,
+		},
+		{
+			note: "erase result",
+			ptr: maskRule{
+				OP:   maskOPRemove,
+				Path: "/result",
+			},
+			event: `{"result": "foo"}`,
+			exp:   `{"erased": ["/result"]}`,
+		},
+		{
+			note: "upsert result",
+			ptr: maskRule{
+				OP:    maskOPUpsert,
+				Path:  "/result",
+				Value: "upserted",
+			},
+			event: `{"result": "foo"}`,
+			exp:   `{"masked": ["/result"], "result": "upserted"}`,
+		},
+		{
+			note: "erase undefined input",
+			ptr: maskRule{
+				OP:   maskOPRemove,
+				Path: "/input/foo",
+			},
+			event: `{}`,
+			exp:   `{}`,
+		},
+		{
+			note: "upsert undefined input",
+			ptr: maskRule{
+				OP:   maskOPUpsert,
+				Path: "/input/foo",
+			},
+			event: `{}`,
+			exp:   `{}`,
+		},
+		{
+			note: "erase undefined result",
+			ptr: maskRule{
+				OP:   maskOPRemove,
+				Path: "/result/foo",
+			},
+			event: `{}`,
+			exp:   `{}`,
+		},
+		{
+			note: "upsert undefined result",
+			ptr: maskRule{
+				OP:   maskOPUpsert,
+				Path: "/result/foo",
+			},
+			event: `{}`,
+			exp:   `{}`,
+		},
+		{
+			note: "erase undefined node",
+			ptr: maskRule{
+				OP:   maskOPRemove,
+				Path: "/input/foo",
+			},
+			event: `{"input": {"bar": 1}}`,
+			exp:   `{"input": {"bar": 1}}`,
+		},
+		{
+			note: "upsert undefined node with nil value",
+			ptr: maskRule{
+				OP:   maskOPUpsert,
+				Path: "/input/foo",
+			},
+			event: `{"input": {"bar": 1}}`,
+			exp:   `{"input": {"bar": 1, "foo": null}, "masked": ["/input/foo"]}`,
+		},
+		{
+			note: "upsert undefined node with a value",
+			ptr: maskRule{
+				OP:    maskOPUpsert,
+				Path:  "/input/foo",
+				Value: "upserted",
+			},
+			event: `{"input": {"bar": 1}}`,
+			exp:   `{"input": {"bar": 1, "foo": "upserted"}, "masked": ["/input/foo"]}`,
+		},
+		{
+			note: "erase undefined node-2",
+			ptr: maskRule{
+				OP:   maskOPRemove,
+				Path: "/input/foo/bar",
+			},
+			event: `{"input": {"foo": 1}}`,
+			exp:   `{"input": {"foo": 1}}`,
+		},
+		{
+			note: "upsert undefined node-2",
+			ptr: maskRule{
+				OP:   maskOPUpsert,
+				Path: "/input/foo/bar",
+			},
+			event: `{"input": {"foo": 1}}`,
+			exp:   `{"input": {"foo": 1}}`,
+		},
+		{
+			note: "erase: undefined object: missing key",
+			ptr: maskRule{
+				OP:   maskOPRemove,
+				Path: "/input/foo/bar/baz",
+			},
+			event: `{"input": {"foo": {}}}`,
+			exp:   `{"input": {"foo": {}}}`,
+		},
+		{
+			note: "upsert: undefined object: missing key",
+			ptr: maskRule{
+				OP:   maskOPUpsert,
+				Path: "/input/foo/bar/baz",
```

Comment:
If we had provided a value, would this create an entry in the modified input ?

Reply:
It would not, since the fully qualified json. pointer to `"/input/foo/bar/baz"` does not exist.